estimatey-api/Estimatey.Infrastructure/DevOps/WorkItemSynchronizationService.cs

```diff
@@ -3,18 +3,19 @@
 using Estimatey.Infrastructure.Persistence;
 using Microsoft.EntityFrameworkCore;
 using Microsoft.Extensions.Logging;
+using System.Collections.Concurrent;
 using System.Text.Json;
 
 namespace Estimatey.Infrastructure.DevOps;
 
 internal class WorkItemSynchronizationService
 {
-    private const int PauseBetweenSyncsMilliseconds = 8000;
-
     private readonly ILogger _logger;
     private readonly DevOpsClient _devOpsClient;
     private readonly ApplicationDbContext _dbContext;
 
+    private readonly ConcurrentDictionary<int, SemaphoreSlim> _semaphores = new();
```

Comment:
Does it matter if this `WorkItemSynchronizationService` and `LinksSynchronizationService` are working on the same project at the same time?

If so then think you'd need to share the `ConcurrentDictionary` between them

Reply:
Same with the Float syncing

Reply:
So float syncing can be completely independant I think.  Re the LinksSyncing and WorkItems Syncing I'm not sure but it is important that work items are synced before links otherwise the link syncer might come across work items which aren't known to the system yet.  There is a also a small chance that work is created and linked between the work item syncing and the link syncing but think chances are small.  Long story short, I think this is fine for now.